### DISCUSSION TITLE
A list of experimental options

### DISCUSSION THREAD

**Discussion by xxczaki** [vercel/next.js#12054] [Help] [answered]
Is it available somewhere?

**Comment by xxczaki** [comment, marked as answer]
**Update July 2021:** If anyone is looking for the list of experimental and "future" options, they are now located in this file:

https://github.com/vercel/next.js/blob/f9795fdd2665369b60a153fbb6231f478e4a7d08/packages/next/server/config-shared.ts

---

Nevermind, I found it: https://github.com/zeit/next.js/blob/450d4bd0f32a042fd452c81bc3850ec31306eab3/packages/next/next-server/server/config.ts#L38

**Reply by kartikag01** [reply]
its very hard to track which are experimental features and which are moved to production on what version,
i have seen people using some options in `experimental` object which are already moved to production, because there is no clear picture in documentation.
If anyone can find something regarding these, its in git issues or discussion.

**Reply by mrmartineau** [reply]
it would also be nice to know what exactly each of them do..

**Reply by leerob** [reply]
There is no documentation around experimental features, because they're experimental 😄 As soon as they're ready for wider adoption, they're moved into the official docs. 

The best source for understanding what they do is to look at the source code!

**Reply by AngeloAnolin** [reply]
Got here as I am getting some warning about some experimental feature I am using in my app.

While admittedly that documentation is not warranted about experimental features, I think this merits reconsideration given that they are part of the whole NextJS framework.

Having some form of documentation / listing of available experimental features can actually lead to better adoption and clarity on which features can be prioritized as well as gather more involvement on how these features will evolve over time.

While admittedly we would want involvement on a deeper technical basis (i.e. traversing the source code to determine these experimental features), it is very likely that 99% of the framework users may not have time to dig deep into the source and could be missing out on some features, that while experimental, will allow for them not to reinvent the wheel.

**Comment by asmyshlyaev177** [comment]
List here for newer versions https://github.com/vercel/next.js/blob/canary/packages/next/src/server/config-schema.ts
